packages/database/src/modelClasses/Entity.js

```diff
@@ -442,7 +445,13 @@ export class EntityModel extends MaterializedViewLogDatabaseModel {
       );
       return Promise.all(relations.map(async r => r.getData()));
     });
-    return Promise.all(relationData.map(async r => this.generateInstance(r)));
+
+    const uniqueEntities = entityArray => {
+      const entitiesById = keyBy(entityArray, 'id');
+      return Object.values(entitiesById);
+    };
+
+    return Promise.all(uniqueEntities(relationData).map(async r => this.generateInstance(r)));
```

Comment:
Bug fix here! Previously if the requested entities had overlapping relatives, we'd return a relative for each entity, leading to duplicates in the responses from entity-server.